metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/MetricCache.java

```diff
@@ -136,6 +154,25 @@ MetricTimeSeries getMetricTimeSeriesRecord(String tenantId, MetricSelection metr
     return metricCache.getIfPresent(Pair.of(tenantId, metricSelection));
   }
 
+  private void recordCacheSize(String tenantId) {
+    AtomicInteger emptyCacheSize = new AtomicInteger(0);
+    cacheSizeGauge
+        .computeIfAbsent(
+            tenantId,
+            k ->
+                PlatformMetricsRegistry.registerGauge(
```

Comment:
We have `PlatformMetricsRegistry.registerCache` this function. Can we use that instead of we calc explicitly?

Reply:
I think we can. Need to check what all metrics are recorded. But the procedure would be same and ig since we are only interested in cache size both would be similar.

Reply:
I would say try that if possible, that way we will have consistent dashboards for cache metrics, and you don't have to calculate size manually.